homeassistant/components/ruckus_unleashed/config_flow.py

```diff
@@ -83,18 +90,45 @@ async def async_step_user(
 
     async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
         """Perform reauth upon an API authentication error."""
+        self._reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
         return await self.async_step_reauth_confirm()
 
     async def async_step_reauth_confirm(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Dialog that informs the user that reauth is required."""
-        if user_input is None:
-            return self.async_show_form(
-                step_id="reauth_confirm",
-                data_schema=DATA_SCHEMA,
-            )
-        return await self.async_step_user()
+        assert self._reauth_entry
+        errors = {}
+        if user_input is not None:
```

Comment:
This is ok, but alternatively we could pass the input to the user step and check if `_reauth_entry` is not None after validation, and in that case update the existing entry and abort instead of creating a new config entry.

Reply:
I think what I've done is what you intended.